llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -71,15 +74,17 @@ Type *VPTypeAnalysis::inferScalarTypeForRecipe(const VPInstruction *R) {
   case VPInstruction::FirstOrderRecurrenceSplice:
   case VPInstruction::Not:
   case VPInstruction::ResumePhi:
+  case VPInstruction::CalculateTripCountMinusVF:
+  case VPInstruction::CanonicalIVIncrementForPart:
+  case VPInstruction::AnyOf:
+  case VPInstruction::LogicalAnd:
```

Comment:
Are there any uses that use LogicalAnd for non-bools? I think the intention is to be only used with bools (LogicalAnd is a poison-safe version of regular bitwise AND for bools). I think that matches the meaning of `LogicalAnd` in other parts of LLVM (but different to general AND)

Reply:
It doesn't look like there is, I think I changed in this when trying to hunt down the cause of the other assertions. I'll change it back and add an assert that it's operand is also a bool